promql/parser/prettier_test.go

```diff
@@ -0,0 +1,488 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAggregateExprPretty(t *testing.T) {
+	maxCharactersPerLine = 10
+	inputs := []struct {
+		in, out string
+	}{
+		{
+			in:  `sum(foo)`,
+			out: `sum(foo)`,
+		},
+		{
+			in: `sum by() (task:errors:rate10s{job="s"})`,
+			out: `sum(
+  task:errors:rate10s{job="s"}
+)`,
```

Comment:
This is pretty hard to read in order to verify. Could we do something about it? Maybe (with proper indentation):
```suggestion
			out: string(			
`sum(
  task:errors:rate10s{job="s"}
)`,
),
```

Your call.